File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    let answer = 0;
    let idx = 0;
    while(idx < section.length){
        const start = section[idx];
        const end = start + m - 1;
        answer++;
        
        while(idx < section.length && section[idx] <= end) idx++;
    }
    
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    idx = 0
    while idx < len(section):
        start = section[idx]
        end = start + m - 1
        answer += 1
        
        while idx < len(section) and section[idx] <= end:
            idx += 1
    return answer